orchestrator/graphql/schemas/product.py

```diff
@@ -36,8 +37,22 @@ async def subscriptions(
         sort_by: Union[list[GraphqlSort], None] = None,
         first: int = 10,
         after: int = 0,
-    ) -> Connection[Annotated["SubscriptionType", strawberry.lazy(".subscription")]]:
+    ) -> Connection[Annotated["SubscriptionInterface", strawberry.lazy(".subscription")]]:
         from orchestrator.graphql.resolvers.subscription import resolve_subscriptions
 
         filter_by_with_related_subscriptions = (filter_by or []) + [GraphqlFilter(field="product", value=self.name)]
         return await resolve_subscriptions(info, filter_by_with_related_subscriptions, sort_by, first, after)
+
+
+@strawberry.federation.type(description="Virtual base class for products", keys=["productId"])
+class ProductType(ProductPydantic):
+    @staticmethod
+    def from_pydantic(model: ProductSchema) -> "ProductType":  # type: ignore
+        graphql_model = ProductPydantic.from_pydantic(model)
+        graphql_model.__class__ = ProductType
+        return graphql_model  # type: ignore
+
+
+@strawberry.experimental.pydantic.type(model=ProductModel, all_fields=True)
+class ProductModelGraphql:
+    pass
```

Comment:
What does this provide next to the `ProductPydantic` class? Do we need both?

Reply:
`ProductModelGraphql` is specific for `SubscriptionInterface` and only has product data, it does not have things like `productBlocks`